runtime/lua/vim/diagnostic.lua

```diff
@@ -132,8 +136,12 @@ local function prefix_source(diagnostics)
   end, diagnostics)
 end
 
----@private
-local function reformat_diagnostics(format, diagnostics)
+--- Returns a copy of the diagnostics with their message reformatted.
+---
+---@param format string A formatting string.
+---@param diagnostics table List of diagnostics |diagnostic-structure|.
+---@return table[] diagnostics |diagnostic-structure|.
+M.reformat_diagnostics = function (format, diagnostics)
```

Comment:
If(!) this is exposed, it needs a more descriptive name following established naming guidelines.

(It also needs a more descriptive description -- "formatted" _how_?)

Reply:
I'll wait to hear _if_ exposing this functions is okay before further refining their docs.